kubestash/__init__.py

```diff
@@ -239,33 +319,48 @@ def kube_init_secret(args, data):
         generate_key(args, key): base64.b64encode(data[key].encode('utf-8')).decode('utf-8')
         for key in data
     }
-    metadata = kubernetes.client.V1ObjectMeta(name=args.secret)
+    metadata = kubernetes.client.V1ObjectMeta(name=name)
     return kubernetes.client.V1Secret(data=converted_data, type='Opaque', metadata=metadata)
 
 
-def kube_create_secret(args, data):
+def kube_create_secret(args, namespace, secret, data):
     """ Creates a Kubernetes secret. Returns the api response from Kubernetes."""
     # https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/docs/CoreV1Api.md#create_namespaced_secret
     kube = get_kube_client()
-    body = kube_init_secret(args, data)
-    return kube.create_namespaced_secret(args.namespace, body)
+    body = kube_init_secret(args, secret, data)
+    return kube.create_namespaced_secret(namespace, body)
 
 
-def kube_replace_secret(args, data):
+def kube_replace_secret(args, namespace, secret, data):
```

Comment:
think there are a few more occurrences `kube_replace_secret` which need to be refactored